_posts/2020-11-30-hhvm-4.86.markdown

```diff
@@ -0,0 +1,40 @@
+---
+title: "HHVM 4.86"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.86 is released! This release marks the end of support for 4.79;
+HHVM 4.80&ndash;4.84 remain supported (4.85 was skipped),
+as does the 4.56 LTS release.
+
+# Breaking Changes
+
+- The built-in function `\preg_last_error` may now return a new value,
+  `\PREG_BAD_REGEX_ERROR`. Various `\preg_...` functions use this error code
+  when called with an invalid regular expression (previously, they would return
+  `false` without setting an error code).
+- If an `async` function is declared with a non-`Awaitable` return type, the
+  typechecker now raises an error at the position of the return type
+  (previously, it would raise an error at the position of the function name).
+  This doesn't introduce any new errors but may require some `HH_FIXME`s to be
+  moved.
+
+# Future Changes
+
+- Calling a function or method "dynamically" (e.g. calling a function/method
+  whose name is stored in a variable) will become more restricted in the future.
+  - The function/method must have the `<<__DynamicallyCallable>>` attribute.
+  - The reference to the function/method to be called must be created using
+    `\HH\dynamic_fun($function_name)` or
+    `\HH\dynamic_class_meth($class_name, $method_name)`.
```

Comment:
no version for instance methods?

Reply:
Good question, I don't see any in the `hhi`s.

@periodic1236 what's the correct way to dynamically call a non-static method?

Reply:
Looks like that's still a work in progress, I'll note that.